### PR TITLE
Jordi/cf 25 slack message contents should be updated

### PR DESCRIPTION
**Pull request by jordiup**

### CODE REVIEW

File: pkg/access/reviewer.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ import (
 type Reviewer struct {
 	ReviewerID string `json:"reviewerId" dynamodbav:"reviewerId"`
 	// Request is the associated request.
-	Request Request `json:"request" dynamodbav:"request"`
+	Request        Request `json:"request" dynamodbav:"request"`
+	SlackMessageID string  `json:"slackMessageId" dynamodbav:"slackMessageId"`
```
Comment (JoshuaWilkes) [inline pkg/access/reviewer.go]: I was thinking some more about this and following our other struct design patterns I think we should make this as below.

It just means that we can expand this in future and know that these are all notifications

``` go

type Notifications struct {
    // if slack is in use, slack message ID should be populated when this has been notified
    SlackMessageID *string `json:"slackMessageId" dynamodbav:"slackMessageId"`
}

type Reviewer struct {
	ReviewerID string `json:"reviewerId" dynamodbav:"reviewerId"`
	// Request is the associated request.
	Request        Request `json:"request" dynamodbav:"request"`
        Notifications Notifications `json:"notifications" dynamodbav:"notifications"`

}

```

File: pkg/notifiers/slack/handle_request.go [hunk 2]
```diff
@@ -99,23 +99,139 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
...
-			n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+			_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
 		}
 	case gevent.RequestApprovedType:
 		msg := fmt.Sprintf("Your request to access *%s* has been approved. Hang tight - we're provisioning the access now and will let you know when it's ready.", ruleQuery.Result.Name)
 		fallback := fmt.Sprintf("Your request to access %s has been approved.", ruleQuery.Result.Name)
-		n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+		_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
+
+		// Loop over the request reviewers
```
Comment (JoshuaWilkes) [inline pkg/notifiers/slack/handle_request.go:136]: Seeing this code block repeated 3 times, makes me think it should be split out into a function?

File: pkg/notifiers/slack/handle_request.go [hunk 3]
```diff
@@ -99,23 +99,139 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
...
+		for _, rev := range reviewers.Result {
+			err := n.UpdateSlackMessage(ctx, slackClient, rev, req, rule, userQuery.Result)
+			if err != nil {
+				log.Errorw("failed to update slack message", "user", rev, zap.Error(err))
+			}
+		}
+
+	case gevent.RequestCancelledType:
+		/*
+			TODO: Decide how the requestor should be notified of the cancellation.
```
Comment (JoshuaWilkes) [inline pkg/notifiers/slack/handle_request.go]: I don;'t think we need to notify the requestor of a cancelled request

File: pkg/notifiers/slack/handle_request.go [hunk 4]
```diff
@@ -99,23 +99,139 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
...
+				}(usr)
 			}
+
 			wg.Wait()
 		} else {
 			//Review not required
 			msg := fmt.Sprintf(":white_check_mark: Your request to access *%s* has been automatically approved. Hang tight - we're provisioning the role now and will let you know when it's ready.", ruleQuery.Result.Name)
 			fallback := fmt.Sprintf("Your request to access %s has been automatically approved.", ruleQuery.Result.Name)
-			n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+			_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
```
Comment (JoshuaWilkes) [inline pkg/notifiers/slack/handle_request.go:129]: Was the existing code for this broken?
Seems like you have swapped and email for an id?

File: pkg/notifiers/slack/handle_request.go [hunk 5]
```diff
@@ -99,23 +99,139 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
...
 		}
 	case gevent.RequestApprovedType:
 		msg := fmt.Sprintf("Your request to access *%s* has been approved. Hang tight - we're provisioning the access now and will let you know when it's ready.", ruleQuery.Result.Name)
 		fallback := fmt.Sprintf("Your request to access %s has been approved.", ruleQuery.Result.Name)
-		n.SendDMWithLogOnError(ctx, slackClient, log, userQuery.Result.Email, msg, fallback)
+		_ = n.SendDMWithLogOnError(ctx, slackClient, log, req.RequestedBy, msg, fallback)
```
Comment (JoshuaWilkes) [inline pkg/notifiers/slack/handle_request.go:134]: same as above, was this broken in main?

**Review by JoshuaWilkes — CHANGES_REQUESTED** [review]
I have asked some questions about the existing code and suggested a tidy up

File: pkg/notifiers/slack/handle_request.go (hunk 3 above)
Comment (jordiup) [reply pkg/notifiers/slack/handle_request.go]: Agreed, have removed it

**Comment by JoshuaWilkes** [comment]
@jordiup coudl you please also update this section of teh docs https://docs.commonfate.io/granted-approvals/configuration/events

**Comment by jordiup** [comment]
> @jordiup coudl you please also update this section of teh docs https://docs.commonfate.io/granted-approvals/configuration/events

https://github.com/common-fate/docs/pull/55

**Review by JoshuaWilkes — APPROVED** [review]

**Pull request merged**